docs/software-architecture-and-design.md

```diff
@@ -472,6 +475,30 @@ The most important thing to know about editing the [Visualizer](#visualizer) in
 
 More information about defining custom widgets in [QtCreator](https://doc.qt.io/qtcreator/creator-using-qt-designer.html) can be found [here](https://doc.qt.io/qt-5/designer-using-custom-widgets.html).
 
+## Qt Best Practices
+Although this will focus on Qt-specific examples, these principles generally apply to all GUI design and implementation. They are really just Software Engineering principles applied to GUIs.
+
+### Build A Hierarchy
+Qt is designed to handle hierarchy. It has [an extensive and robust system for maintaining Object Trees and parent/child relationships between components](https://doc.qt.io/qt-5/objecttrees.html). Much like regular code, GUIs should be created in a logical hierarchy to make organization and re-use easier.
+
+Do **not** lay out all your widgets in a single layer like this:
+![Bad Qt Widget Layout](images/qt_layout_bad_example.png)
```

Comment:
oof

Reply:
lol sorry I didn't want to make my own example